chapter03/src/test/java/com/github/sebin/chapter03/BankConfig.java

```diff
@@ -0,0 +1,40 @@
+package com.github.sebin.chapter03;
+import java.util.Calendar;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.github.sebin.chapter03.Account;
+import com.github.sebin.chapter03.service.BankService;
+import com.github.sebin.chapter03.service.impl.XBank;
+
+@Configuration
+public class BankConfig {
+
+	@Bean
+	public User user() {
+		return new User(1, "xbxlnv5", "123456", "My Name");
+	}
+
+
+	@Bean
+	public Account fromAccount() {
+		return new Account(1, user(), "Testing account 1", 10, 100000);
```

Comment:
This is very interesting, I come across this as well, that `user()` actually returns a `new` User, but looks like somehow Spring must figure out other `user()` calls are referencing to the same User instance